src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/migration/BookingStatusMigrationJob.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.migration
+
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Slice
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.BookingStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BookingEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BookingRepository
+import javax.persistence.EntityManager
+
+class BookingStatusMigrationJob(
+  private val bookingRepository: BookingRepository,
+  private val entityManager: EntityManager,
+  private val pageSize: Int,
+) : MigrationJob() {
+  private val log = LoggerFactory.getLogger(this::class.java)
+  override val shouldRunInTransaction = true
+
+  override fun process() {
+    log.info("Starting Booking Migration process...")
+
+    var page = 1
+    var hasNext = true
+    var slice: Slice<BookingEntity>
+
+    while (hasNext) {
+      log.info("Getting page $page")
+      slice = bookingRepository.findAllBookingWithNullStatus(PageRequest.of(0, pageSize))
+      slice.content.forEach {
+        setStatus(it)
+      }
+      entityManager.clear()
+      hasNext = slice.hasNext()
+      page += 1
+    }
+    log.info("Booking Status Migration process complete!")
+  }
+
+  private fun setStatus(booking: BookingEntity) {
```

Comment:
It'd be good if we could update this to handle CAS1 Bookings too - I think we'd be fine to have bookings default to `provisional` for now as we'll be introducing the concept of confirmations soon

Reply:
I left the migration of CAS1 and CAS2 booking intentionally, because i dont have the requirement and I dont want to assume and create wrong status. 
I would request you to add story in respective team to migrate the status as per the requirement.

Reply:
Yeah, that's totally fair 👍